### ISSUE TITLE
OvO tier 2 test test_src/fortran/hierarchical_parallelism/memcopy-real-loop_pragma/target__teams__loop.F90 has programming issue

### ISSUE THREAD

**Issue by erkle131** [TApplencourt/OvO#28]
Hello,

I'm a QA engineer at Intel and one of our devs identified a programming issue in the following test.

The test target__teams__loop.F90 has this construct:
```

  !$OMP TARGET map(to: src) map(from: dst) private(idx)
  !$OMP TEAMS private(idx)
  !$OMP LOOP
  DO i0 = 1, N0
    idx = i0-1+1
    dst(idx) = src(idx)
  END DO
  !$OMP END TEAMS
  !$OMP END TARGET
```

The "private(idx)" clause should be moved from the OMP TEAMS directive to the OMP LOOP directive.

The reasoning is because of this quote from the OpenMP 5.1 manual, in the section for Data-Sharing Attribute Rules.
```

23    • In a parallel construct, if no default clause is present, these variables are shared.
24    • For constructs other than task generating constructs, if no default clause is present, these
25        variables reference the variables with the same names that exist in the enclosing context.
```

The LOOP will mark the variable "idx" as SHARED.  However, it needs to be PRIVATE in order for the program to succeed.

**Comment by TApplencourt** [comment]
Thanks, I will fix it asap!
Do you know if putting  `shared` only on the `!$OMP LOOP` will work?

**Comment by erkle131** [comment]
Hi @TApplencourt, did you mean "put private" instead of "shared"? The below code snippet worked.
```

!$OMP TARGET map(to: src) map(from: dst) private(idx)
!$OMP TEAMS
!$OMP LOOP private(idx)
  DO i0 = 1, N0
    idx = i0-1+1
    dst(idx) = src(idx)
  END DO
!$OMP END TEAMS
!$OMP END TARGET
```

**Comment by TApplencourt** [comment]
Perfect thanks. I was thinking:
```
!$OMP TARGET map(to: src) map(from: dst)
!$OMP LOOP private(idx)
  DO i0 = 1, N0
    idx = i0-1+1
    dst(idx) = src(idx)
  END DO
!$OMP END TEAMS
!$OMP END TARGET
```
but your test is similar. 

For implementation reasons, I will put private everywhere (it's simpler to do), but good to know :)

**Comment by TApplencourt** [comment]
Fixed by 3be93d52d0fabeb3986b0b3ee58ef5c8ad92b997 . 
As usual thanks for the report!
